common/include/pcl/common/impl/io.hpp

```diff
@@ -51,13 +51,8 @@ pcl::getFieldIndex (const pcl::PointCloud<PointT> &,
                     const std::string &field_name, 
                     std::vector<pcl::PCLPointField> &fields)
 {
-  fields.clear ();
-  // Get the fields list
-  pcl::for_each_type<typename pcl::traits::fieldList<PointT>::type>(pcl::detail::FieldAdder<PointT>(fields));
-  for (std::size_t d = 0; d < fields.size (); ++d)
-    if (fields[d].name == field_name)
-      return (static_cast<int>(d));
-  return (-1);
+  return getFieldIndex<PointT>(std::forward<decltype(field_name)>(field_name),
```

Comment:
Could you please explain this?

Reply:
The function being called is right below this. Since the first variable is being ignored, no need to duplicate code. I wanted to mark this as deprecated because it doesn't make sense but maybe later

Reply:
I'm sending the parameters to the function right below this one. This function has a dead variable (which made me almost add deprecated because it feels so), so I decided to keep the live code in the function with least parameters (and same name, and function body)

Reply:
OK, got that. But what is `std::forward<decltype(...)>` for? Wouldn't it work without?

Reply:
`std::forward` allows a temporary to remain one and not become a reference. Since the `field_name` might be just constant char array (like "x"), the compiler might create std::string and pass its reference instead of passing it directly.

To be frank, it might not affect anything here, but it's a reflex to forward arguments using `std::forward` to prevent a copy (and for correct selection of overloads, lest someone take reference to a temporary because we passed a reference variable, not a temporary)

Reply:
> Since the field_name might be just constant char array (like "x"),

I don't think I agree with you here. `field_name` **is** a `const std::string&`, no matter how the function was called. Yes, it may be bound to a temporary constructed out of a C-string, but still the type of `field_name` variable is fixed to be `const std::string&`. Therefore, `decltype(field_name)` === `const std::string&`.